store/tikv/2pc_test.go

```diff
@@ -379,3 +379,33 @@ func (s *testCommitterSuite) TestPrewritePrimaryKeyFailed(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(v, BytesEquals, []byte("a3"))
 }
+
+func (s *testCommitterSuite) TestWrittenKeysOnConflict(c *C) {
+	// This test checks that when there is a write conflict, written keys is collected,
+	// so we can use it to clean up keys.
+	region, _ := s.cluster.GetRegionByKey([]byte("x"))
+	newRegionID := s.cluster.AllocID()
+	newPeerID := s.cluster.AllocID()
+	s.cluster.Split(region.Id, newRegionID, []byte("y"), []uint64{newPeerID}, newPeerID)
+	for i := 0; i < 100; i++ {
+		txn1 := s.begin(c)
+
+		txn2 := s.begin(c)
+		txn2.Set([]byte("x1"), []byte("1"))
+		commiter2, err := newTwoPhaseCommitter(txn2, 2)
+		c.Assert(err, IsNil)
+		err = commiter2.execute(context.Background())
+		c.Assert(err, IsNil)
+
+		txn1.Set([]byte("x1"), []byte("1"))
+		txn1.Set([]byte("y1"), []byte("2"))
+		commiter1, err := newTwoPhaseCommitter(txn1, 2)
+		c.Assert(err, IsNil)
+		err = commiter1.execute(context.Background())
+		c.Assert(err, NotNil)
+		commiter1.mu.RLock()
+		writtenKeyLen := len(commiter1.mu.writtenKeys)
+		commiter1.mu.RUnlock()
+		c.Assert(writtenKeyLen, Equals, 1)
```

Comment:
How do we know the keys are cleaned up?

Reply:
We fix the problem written keys is empty, So.
The unit test only need to verify that the written keys are collect.